src/DIRAC/TransformationSystem/scripts/dirac_transformation_update_derived.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+"""Move files that are Unused or MaxReset from a parent production to its
+derived production. The argument is a list of productions: comma separated list
+of ranges (a range has the form p1:p2)"""
+
+import DIRAC
+from DIRAC import gLogger
+from DIRAC.Core.Base.Script import Script
+from DIRAC.TransformationSystem.Client.TransformationClient import TransformationClient
+
+
+@Script()
+def main():
+
+    Script.registerSwitch("", "NoReset", "Don't reset the MaxReset files to Unused (default is to reset)")
+    Script.parseCommandLine(ignoreErrors=True)
+
+    transClient = TransformationClient()
+
+    resetUnused = True
+    switches = Script.getUnprocessedSwitches()
+    for switch in switches:
+        if switch[0] == "NoReset":
+            resetUnused = False
+    args = Script.getPositionalArgs()
+
+    if not args:
+        gLogger.error("Specify transformation number...")
+        DIRAC.exit(0)
+
+    ids = args[0].split(",")
+    idList = []
+    for transId in ids:
+        r = transId.split(":")
+        if len(r) > 1:
+            for i in range(int(r[0]), int(r[1]) + 1):
+                idList.append(i)
+        else:
+            idList.append(int(r[0]))
+
+    for prod in idList:
+        res = transClient.getTransformation(prod, extraParams=True)
+        if not res["OK"]:
+            gLogger.error("Error getting transformation", f"{prod}: {res['Message']}")
+        else:
+            res = transClient.moveFilesToDerivedTransformation(res["Value"], resetUnused)
+            if not res["OK"]:
+                gLogger.error("Error updating a derived transformation", f"{prod}: {res['Message']}")
+            else:
```

Comment:
```suggestion
            continue
```
even less indentations